File: 392-is-subsequence/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        n1, n2 = len(s), len(t)
        i, j = 0, 0

        while(i < n1 and j < n2):
            if(s[i] == t[j]):
                i += 1
            j += 1

        if(i == n1):
            return True
        else:
            return False

File: 392-is-subsequence/is-subsequence.cpp
#define ll long long
#define vi vector<int>
#define vll vector<ll>
#define vs vector<string>
#define mpi map<int, int>
#define mps map<string, int>
#define pb push_back
#define all(x) (x).begin(), (x).end()

const int MOD = 1e9 + 7;

class Solution {
public:
    bool isSubsequence(string s, string t) {
        int n = s.length(), m = t.length();
        int x = 0;

        for (int i = 0; i < m and x < n; i++) {
            if (s[x] == t[i]) {
                x++;
            }
        }

        return (x == n);
    }
};